upload.go

```diff
@@ -311,17 +323,33 @@ func (ua *uploadAction) uploadBuild() error {
 	resp, err := client.Do(req)
 
 	if err != nil {
-		return fmt.Errorf("Unable to upload build to Waldo, error: %v, url: %s", err, url)
+		return retryAllowed, fmt.Errorf("Unable to upload build to Waldo, error: %v, url: %s", err, url)
 	}
 
 	dumpResponse(ua.userVerbose, resp, true)
 
 	defer resp.Body.Close()
 
-	return ua.checkBuildStatus(resp)
+	return retryAllowed && shouldRetry(resp), ua.checkBuildStatus(resp)
+}
+
+func (ua *uploadAction) uploadBuildWithRetry() error {
```

Comment:
nit: I want to say this could be made common code and use the new Go generics but maybe it's overkill :)

Reply:
Yeah, that might be a bit too much this time around.